File: Adorable/app/src/main/java/merryweather/com/adorable/di/modules/ActivityModule.java
package merryweather.com.adorable.di.modules;

import android.app.Activity;
import android.content.Context;

import dagger.Module;
import dagger.Provides;
import merryweather.com.adorable.di.scopes.PerActivity;

/**
 * Created by S on 19.04.2018.
 */

@Module
public class ActivityModule {

    private final Activity activity;

    public ActivityModule(Activity activity) {
        this.activity = activity;
    }

    @Provides
    @PerActivity
    Context provideContext() {
        return activity;
    }

}

File: Tech45/app/src/main/java/merryweather/com/tech45/di/modules/ActivityModule.kt
package merryweather.com.tech45.di.modules

import android.app.Activity
import android.content.Context

import dagger.Module
import dagger.Provides
import merryweather.com.tech45.di.scopes.PerActivity

/**
 * Created by S on 19.04.2018.
 */

@Module
class ActivityModule(private val activity: Activity) {

    @Provides
    @PerActivity
    internal fun provideContext(): Context {
        return activity
    }

}